pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -103,21 +109,16 @@ func (p *Plugin) Configure(ctx context.Context, req *plugin.ConfigureRequest) (*
 	p.keyPrefix = config.KeyPrefix
 	p.kmsClient, err = p.hooks.newClient(ctx, config)
 	if err != nil {
-		return nil, kmsErr.New("failed to create KMS client: %v", err)
+		return nil, newErrorf(codes.Internal, "failed to create KMS client: %v", err)
 	}
 
 	p.log.Debug("Fetching key aliases from KMS")
-	var nextMarker *string
-	for {
-		nextMarker, err = p.fetchAliasesPage(ctx, nextMarker)
-		if err != nil {
-			return nil, err
-		}
-		if nextMarker == nil {
-			break
-		}
+	err = p.buildCache(ctx)
```

Comment:
This action is destructive to the existing cache operations and can fail if it encounters a malformed entry part way through. I'd suggest we refactor to have a stateless function that obtains and returns validated entries so that the operation that actually mutates state is no-fail.